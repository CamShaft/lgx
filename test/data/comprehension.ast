#{
  type => comprehension,
  children => #{
    assignment => #{
      type => assign,
      value => 'User',
      children => #{
        0 => #{
          type => list,
          children => #{
            0 => #{
              type => literal,
              value => <<"1">>
            },
            1 => #{
              type => literal,
              value => <<"2">>
            },
            2 => #{
              type => literal,
              value => <<"3">>
            },
            3 => #{
              type => literal,
              value => <<"4">>
            },
            4 => #{
              type => literal,
              value => <<"5">>
            }
          }
        }
      }
    },
    filters => [
      #{
        type => call,
        value => {users, is_allowed},
        children => #{
          0 => #{
            type => variable,
            value => 'User'
          }
        }
      }
    ],
    body => [
      #{
        type => assign,
        value => 'User2',
        children => #{
          0 => #{
            type => call,
            value => {users, fetch_user},
            children => #{
              0 => #{
                type => variable,
                value => 'User'
              }
            }
          }
        }
      },
      #{
        type => call,
        value => {maps, get},
        children => #{
          0 => #{
            type => literal,
            value => name
          },
          1 => #{
            type => variable,
            value => 'User2'
          }
        }
      }
    ]
  }
}.