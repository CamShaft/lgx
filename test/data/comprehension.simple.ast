#{
  type => comprehension,
  children => #{
    assignment => #{
      type => assign,
      value => 'User',
      children => #{
        0 => #{
          type => list,
          children => #{
            0 => #{
              type => literal,
              value => <<"1">>
            },
            1 => #{
              type => literal,
              value => <<"2">>
            },
            2 => #{
              type => literal,
              value => <<"3">>
            },
            3 => #{
              type => literal,
              value => <<"4">>
            },
            4 => #{
              type => literal,
              value => <<"5">>
            }
          }
        }
      }
    },
    expression => #{
      type => call,
      value => {maps, get},
      children => #{
        0 => #{
          type => literal,
          value => name
        },
        1 => #{
          type => variable,
          value => 'User'
        }
      }
    }
  }
}.