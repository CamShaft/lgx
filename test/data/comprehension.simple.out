[{maps, get, [name, <<"1">>]},
 {maps, get, [name, <<"2">>]},
 {maps, get, [name, <<"3">>]},
 {maps, get, [name, <<"4">>]},
 {maps, get, [name, <<"5">>]}].